opencti-platform/opencti-graphql/src/graphql/sseMiddleware.js

```diff
@@ -598,23 +599,34 @@ const createSseMiddleware = () => {
                   } else if (!isStixDomainObjectContainer(elementType)) { // Update but not visible - entity type
                     // If entity is not a container, it can be part of a container that is authorized by the filters
                     // If it's the case, the element must be published
-                    const filters = {
-                      mode: 'and',
-                      filters: [{ key: [buildRefRelationKey(RELATION_OBJECT)], values: [elementInternalId] }],
-                      filterGroups: [],
-                    };
-                    const args = { connectionFormat: false, filters };
-                    const containers = await listEntities(context, user, [ENTITY_TYPE_CONTAINER], args);
                     let isContainerMatching = false;
-                    for (let containerIndex = 0; containerIndex < containers.length; containerIndex += 1) {
-                      const container = containers[containerIndex];
-                      const stixContainer = convertStoreToStix(container);
-                      const containerMatch = await isStixMatchFilterGroup(context, user, stixContainer, streamFilters);
-                      if (containerMatch) {
-                        isContainerMatching = true;
-                        break;
+                    const safeFilters = structuredClone(streamFilters); // Safe clone to be compliant with async iteration usage
+                    const listToCallback = async (containers) => {
+                      const containersIds = containers.map((c) => c.internal_id);
+                      const containersData = await stixLoadByIds(context, req.user, containersIds);
+                      for (let containerIndex = 0; containerIndex < containersData.length; containerIndex += 1) {
+                        const stixContainer = containersData[containerIndex];
+                        const containerMatch = await isStixMatchFilterGroup(context, user, stixContainer, safeFilters);
+                        if (containerMatch) {
+                          isContainerMatching = true;
+                          break;
+                        }
                       }
-                    }
+                      return !isContainerMatching; // If match is found, stop the listing process
+                    };
+                    await listAllEntities(context, user, [ENTITY_TYPE_CONTAINER], {
+                      first: CONTAINER_FINDER_BATCH_RESOLUTION,
+                      baseData: true,
+                      orderBy: 'updated_at', // Get container by newly updated
+                      orderMode: 'desc',
+                      connectionFormat: false,
+                      filters: {
```

Comment:
couldn't we add `streamFilters` here to query only elements filtered by this stream ? we wouldn't even need to read the results, only count.

Reply:
We could and we should. Just commit a new approach following this remark. Thanks